sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1216,8 +1216,12 @@ object SQLConf {
       .createWithDefault(true)
 
   val ENABLE_FALL_BACK_TO_HDFS_FOR_STATS = buildConf("spark.sql.statistics.fallBackToHdfs")
-    .doc("If the table statistics are not available from table metadata enable fall back to hdfs." +
-      " This is useful in determining if a table is small enough to use auto broadcast joins.")
+    .doc("This flag is effective only if it is Hive table. When true, it will fall back to HDFS " +
+      "if the table statistics are not available from table metadata. This is useful in " +
+      "determining if a table is small enough to use auto broadcast joins. " +
+      "For non-partitioned data source table, it will be automatically recalculated if table " +
+      "statistics are not available. For partitioned data source table, It is " +
+      s"'${DEFAULT_SIZE_IN_BYTES.key}' if table statistics are not available.")
     .booleanConf
```

Comment:
cc @shahidki31

Reply:
Thanks @wangyum . Looks good. Shall we add it in the configuration.md also. It seems these configs are not there?

Reply:
Hi, @wangyum . We had better minimize the patch diff always. Please keep the original location of this conf. I don't see any difference for the following three lines.
> val ENABLE_FALL_BACK_TO_HDFS_FOR_STATS = buildConf("spark.sql.statistics.fallBackToHdfs")
> .booleanConf
> .createWithDefault(false)

Reply:
This is because we added [DEFAULT_SIZE_IN_BYTES](https://github.com/apache/spark/pull/24715/files#diff-9a6b543db706f1a90f790783d6930a13R1233) to `ENABLE_FALL_BACK_TO_HDFS_FOR_STATS`.  So we need to move `DEFAULT_SIZE_IN_BYTES` before `ENABLE_FALL_BACK_TO_HDFS_FOR_STATS `, otherwise:
```
[error] [warn] /home/jenkins/workspace/SparkPullRequestBuilder/sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala:1224: Reference to uninitialized value DEFAULT_SIZE_IN_BYTES
[error] [warn]       s"'${DEFAULT_SIZE_IN_BYTES.key}' if table statistics are not available.")
[error] [warn] 
[warn] 8 warnings found
```

Reply:
@shahidki31 Yes. Running the `SET -v` command will show the entire list of the SQL configuration:
http://spark.apache.org/docs/latest/configuration.html#spark-sql